scripts/sysman/engine.yml

```diff
@@ -69,11 +69,13 @@ name: $s_engine_stats_t
 members:
     - type: uint64_t
       name: "activeTime"
-      desc: "[out] Monotonic counter for time in microseconds that this resource is actively running workloads."
+      desc: |
+            [out] Monotonic counter where the resource is actively running workloads.
```

Comment:
what is the driver returning today, us or raw counters?

Reply:
currently it is us, but data is fundamentally used only to calculate % utilization per formula. so obfuscating units from spec pov allows implementation flexibility to report raw data in case underlying HW is counting at different frequencies than us/ms.

Reply:
wouldnt this cause a problem for customers, since they are assuming this is returned in us?

Reply:
as far as we know customers are not using the fact that it's in us in any way currently. i.e, they just use the formula to calculate % utilization

Reply:
@aravindksg could you share details on that formula? doesn't the formula assume the value to be on us?

Reply:
formula is from here: https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-engine-stats-t

```
Percent utilization is calculated by taking two snapshots (s1, s2) and using the equation: 
util = (s2.activeTime - s1.activeTime) / (s2.timestamp - s1.timestamp)
```

so here, doesn't really matter what units we have, we are really interested to just find the ratio. it just matters that `activeTime` and `timestamp` values are relatable so that this calculation works.

Reply:
@aravindksg  you are removing restriction to represent in us, instead this structure could represent in any time either ms or ns or us right ?  But don;t we need to make sure both active time and time stamp should represent in same time unit whether it is sec or ms or us or ns ?

Reply:
@saik-intel regarding why we want to remove time units completely, pasting reply from above:

> obfuscating units from spec pov allows implementation flexibility to report raw data in case underlying HW is counting at different frequencies than us/ms

> don;t we need to make sure both active time and time stamp should represent in same

yes, both these values need to be relatable. I can add statement to that effect. we already have following statement in the spec: `Never take the delta of this timestamp with the timestamp from a different structure since they are not guaranteed to have the same base`. so can apply similar strict wording to enforce that activeTime and timestamp are in same units

Reply:
thanks @aravindksg. 

> so obfuscating units from spec pov allows implementation flexibility to report raw data in case underlying HW is counting at different frequencies than us/ms.

could you elaborate on the need for that flexibility?

>formula is from here: https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-engine-stats-t

right, there's a formula, but the spec doesn't say those fields could be used by the user in other way. here for instance:

> uint64_t activeTime
[out] Monotonic counter for time in microseconds that this resource is actively running workloads.

it doesnt say that the activeTime can only be used on that formula. Furthermore, if we are saying that activeTime and timestamp are only useful for the formula, why dont we just return the value of the formula and have the driver implementation calculating it internally?

Reply:
> thanks @aravindksg.
> 
> > so obfuscating units from spec pov allows implementation flexibility to report raw data in case underlying HW is counting at different frequencies than us/ms.
> 
> could you elaborate on the need for that flexibility?
> 

one example would be that the wall clock times could be different in VF vs PF domain. So when PF queries engine stats for all VFs, and if it infers in time (since we currently have "microseconds" wording) then it would be unclear to user what reference point of time would be (since environments are different)

> > formula is from here: https://spec.oneapi.io/level-zero/latest/sysman/api.html#zes-engine-stats-t
> 
> right, there's a formula, but the spec doesn't say those fields could be used by the user in other way. here for instance:
> 
> > uint64_t activeTime
> > [out] Monotonic counter for time in microseconds that this resource is actively running workloads.
> 
> it doesnt say that the activeTime can only be used on that formula. Furthermore, if we are saying that activeTime and timestamp are only useful for the formula, why dont we just return the value of the formula and have the driver implementation calculating it internally?

yes, implementation could do it, but the calculation is applied for 2 snapshots taken at two different time intervals. So if driver has to do it, then it might need a "timeout" param so that driver can block for that value and take 2 snapshots of engine activity.

Reply:
> Time units are intentionally obfuscated

to

> Time units are implementation specific

Reply:
@jandres742 any rationale for using "implementation specific" wording rather than "obfuscated"? because terming "time units" as implementation specific is unclear IMHO.

Reply:
in terms of specifications, "implementation specific" is more used than "obfuscated". Basically what we are saying is that user shouldn't assume anything about the time units.

@wdamon-intel : what do you think?

Reply:
fixed